spark/src/main/scala/com/metamx/tranquility/spark/BeamRDD.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.metamx.tranquility.spark
+import com.twitter.util.Await
+import org.apache.spark.rdd.RDD
+import scala.reflect.ClassTag
+import com.metamx.common.scala.Logging
+import scala.language.implicitConversions
+
+/**
+  * This class provides any RDD the ability to propagate events to Druid.
+  * @param rdd rdd that should be propagated to druid.
+  */
+class BeamRDD[T: ClassTag](rdd:RDD[T]) extends Logging with Serializable{
+
+  def propagate(beamFactory: BeamFactory[T]) = {
```

Comment:
Thanks a ton for the feedback Gian! I am working on them, will check in the
changes in a little bit.

Best,
Deepika

On Thu, Dec 3, 2015 at 1:12 PM, Gian Merlino <notifications@github.com>
wrote:

> In spark/src/main/scala/com/metamx/tranquility/spark/BeamRDD.scala
> <https://github.com/druid-io/tranquility/pull/58#discussion_r46612541>:
>
> > + * under the License.
> > + */
> > +package com.metamx.tranquility.spark
> > +import com.twitter.util.Await
> > +import org.apache.spark.rdd.RDD
> > +import scala.reflect.ClassTag
> > +import com.metamx.common.scala.Logging
> > +import scala.language.implicitConversions
> > +
> > +/**
> > +  * This class provides any RDD the ability to propagate events to Druid.
> > +  * @param rdd rdd that should be propagated to druid.
> > +  */
> > +class BeamRDD[T: ClassTag](rdd:RDD[T]) extends Logging with Serializable{
> > +
> > +  def propagate(beamFactory: BeamFactory[T]) = {
>
> propagate is ok too I guess, the BeamFactory makes it pretty clear what's
> going on.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/druid-io/tranquility/pull/58/files#r46612541>.
>